pkg/types/chain_reader.go

```diff
@@ -30,7 +33,11 @@ type ChainReader interface {
 	// 		err = cr.GetLatestValue(ctx, bc, "method", ProductParams{Arg:1}, &resp)
 	// 		return
 	//  }
-	GetLatestValue(ctx context.Context, bc BoundContract, method string, params, returnVal any) error
+	GetLatestValue(ctx context.Context, contractName string, method string, params, returnVal any) error
```

Comment:
Is codec now the only option? Or is JSON still available?
Either way, these comments need to be updated.